### PR TITLE
Masking JMS Message Store parameter values using secure-vault

### PR DESCRIPTION
**Pull request by Nipunich**
## Purpose
Implement the Masking JMS Message Store parameter values using secure-vault capability.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java [hunk 1]
```diff
@@ -712,4 +723,17 @@ private String nameString() {
     public void setProducer(MessageProducer producer) {
         this.producer = producer;
     }
+
+    public String resolveQueueName(Object param) {
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:727]: I feel like this method is to resolve all the vault-look ups in the connection string. In that case, why did we name this as 'resolveQueueName'?

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java [hunk 2]
```diff
@@ -119,6 +121,11 @@
 
     private SynapseEnvironment synapseEnvironment;
 
+    /** regex for secure vault expression */
+    private static final String SECURE_VAULT_REGEX = "\\{(wso2:vault-lookup\\('(.*?)'\\))\\}";
+
+    private Pattern vaultLookupPattern = Pattern.compile(SECURE_VAULT_REGEX);
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:127]: Can we use the the variables defined in SecureVaultResolver without re-defining them?

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java (hunk 1 above)
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:727]: I feel like this method is to resolve all the vault-look ups in the connection string. In that case, why did we name this as 'resolveQueueName'?

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java [hunk 3]
```diff
@@ -49,6 +49,8 @@
 import java.util.NoSuchElementException;
 import java.util.Properties;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class JmsStore extends AbstractMessageStore {
```
Comment (arunans23) [inline modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:55]: Are we doing this only for JmsStore? What about other message stores?

**Comment by Nipunich** [comment]
The testing is in progress. Will reopen once completed.

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java (hunk 1 above)
Comment (Nipunich) [reply modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java]: Fixed

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java (hunk 2 above)
Comment (Nipunich) [reply modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java]: Fixed

File: modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java (hunk 3 above)
Comment (Nipunich) [reply modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:55]: At the moment we did this only for JMS Store.

Comment (arunans23) [reply modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:55]: Let's rename the PR title as JMS Message Store.

Comment (Nipunich) [reply modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java:55]: Done

**Pull request merged**